### PR TITLE
add readme

### PR DESCRIPTION
**Pull request by nichanank**

### CODE REVIEW

**Review by jrhite — APPROVED** [review]
This is a really excellent README! 🌟  

The only thing that might be worth adding is kovan deployment info? Thoughts?

I have a PR pending https://github.com/bazaarfinance/bazaarfinance/pull/14 consolidating how hardhat config files and and env files work for diff environments. When you or Pong approve I'll merge to master.

Left some minor comments for you that relate to the env var setup and my PR.

Great stuff!

File: README.md [hunk 1]
```diff
@@ -1,39 +1,82 @@
...
+- Ensure Metamask RPC is set to `http://localhost:8545` and chainID `31337`.
+- Run `yarn start` or npm `npm start` and a new browser window should open up at http://localhost:3000/
 
-Ensure you are useing RPC to http://localhost:8545.
+## ✅ Testing
+- Run unit tests `npx hardhat test`
+- Run integration tests `npx hardhat --config hardhat.integration-test-config.ts test ./test/integration/*.ts` from the main directory
 
-You may also need to set the chainID to 31337 if you are useing Hardhat blockchain development node.
+Since the tests fork mainnet for interaction with AAVE, you'll need to interact with a node service such as [Alchemy](http://alchemyapi.io/), which requires an API Key, for tests to work. See `.env-sample` for instructions on setting up a local `.env` file that stores the mainnet `FORKING_URL`
```
Comment (jrhite) [inline README.md:77]: moved all example env files to env-example-files in https://github.com/bazaarfinance/bazaarfinance/pull/14.

`.env-sample` --> `env-example-files/.env-example`

File: README.md [hunk 2]
```diff
@@ -1,39 +1,82 @@
...
+5. Interest on deposits accrues to the project's `Vault`. Every 30 days, the accrued interest gets allocated to be `recipient`, available to be withdrawn.
+6. If the interest accrued from the sum of deposited funds exceeds the recipient's desired monthly `salary` for that month, additional interest earned for the remainder of that month gets allocated to depositors.
+7. Recipients can `withdraw` a specified `amount` at any time.
+8. Depositors can `withdraw` their principal (and any additional interest) at any time.
+
+Note: The `Vault` has sole access to `mint` and `burn` on its associated `bToken`, as these functions need only be called during `deposit` and `withdraw`
+
+### bToken Exchange Rate
+
+The 1-1 exchange rate only applies to the initial deposit because any subsequent deposits has to take into account (1) the interest already accrued and (2) whether salary goals have been met. Each bToken is convertible into a dynamic quantity of the underlying asset, as depositors deposit and withdraw from the project's `Vault`.
```
Comment (Pongch) [inline README.md:52]: very concise, nicely done

File: README.md [hunk 3]
```diff
@@ -42,13 +85,15 @@ eth_sendRawTransaction
 
 Reset your account in Metamask.
 
-# We ❤️ these **Ethereum** projects:
+## 🚀 Future Goals & TODOs
+- Host immutable footballer data for a given season on IPFS
+- Add ENS capability to resolve human-readable names to Ethereum addresses
+- Implement upgradable design or autodeprecation
+- Add league admin features to the client web app
```
Comment (Pongch) [inline README.md:92]: wait wot lol

File: README.md [hunk 4]
```diff
@@ -42,13 +85,15 @@ eth_sendRawTransaction
 
 Reset your account in Metamask.
 
-# We ❤️ these **Ethereum** projects:
+## 🚀 Future Goals & TODOs
+- Host immutable footballer data for a given season on IPFS
+- Add ENS capability to resolve human-readable names to Ethereum addresses
+- Implement upgradable design or autodeprecation
```
Comment (Pongch) [inline README.md:91]: is `autodeprecation` one word?

File: README.md [hunk 5]
```diff
@@ -1,39 +1,82 @@
-# Get started
+# Bazaar Finance
 
-1. Clone the repo and cd into it `git clone https://github.com/symfoni/hardhat-react-boilerplate.git MyProject && cd MyProject`
-2. Install deps with yarn `yarn` or npm `npm install`
-3. Start hardhat `npx hardhat node --watch`
+Bazaar Finance allows developers to capture value for their work in open-source software though decentralized finance (DeFi) protocols. Thanks to lending protocols, users around the globe now have a new avenue to fund the open-source tools they find valuable. Not by dipping into their wallets, but via interest earned on their savings. Instead of having to actively give lump sums or set up recurring monthly donations, people can allocate a portion of interest they earn, passively and transparently, with the ability to withdraw their principle at any time.
```
Comment (Pongch) [inline README.md:3]: nice intro! do we wanna prioritize 2 additional product values to differentiate us from projects like rDAI:

1. for recipient: they can have a recurring source of income, predictably
2. for depositor: they can earn additional yield, making it superior to simply just holding the underlying asset w/o depositing with Bazaar

File: README.md [hunk 6]
```diff
@@ -1,39 +1,82 @@
...
+2. Individuals who wish to contribute the interest on their savings to fund the project can `deposit` a specified `amount` of the projects `token` into its `Vault`.
+3. The deposit gets automatically swapped to the interest-accruing equivalent on Aave, and is stored in the `Vault`.
+4. The depositor gets minted the project's `bToken`, redeemable for underlying funds when they `withdraw` their principal (and any additional interest) from the `Vault`
+5. Interest on deposits accrues to the project's `Vault`. Every 30 days, the accrued interest gets allocated to be `recipient`, available to be withdrawn.
+6. If the interest accrued from the sum of deposited funds exceeds the recipient's desired monthly `salary` for that month, additional interest earned for the remainder of that month gets allocated to depositors.
+7. Recipients can `withdraw` a specified `amount` at any time.
+8. Depositors can `withdraw` their principal (and any additional interest) at any time.
+
+Note: The `Vault` has sole access to `mint` and `burn` on its associated `bToken`, as these functions need only be called during `deposit` and `withdraw`
```
Comment (Pongch) [inline README.md:48]: do we wanna mention additional note that, for current implementation:

1. `withdraw` will withdraw their entire balance and 2. user
2.  users aren't allow to make transfers of bTokens?

**Review by Pongch — COMMENTED** [review]
great writeup 👏 

my only question is on the football-oriented ending to this otherwise excellent readme.

File: README.md (hunk 6 above)
Comment (nichanank) [reply README.md:48]: good points! added 👍🏼

File: README.md (hunk 3 above)
Comment (nichanank) [reply README.md:92]: 😅 removed

**Pull request merged**